noir-projects/aztec-nr/aztec/src/context/private_context.nr

```diff
@@ -305,16 +326,24 @@ impl PrivateContext {
         assert(args_hash == item.public_inputs.args_hash);
 
         // Assert that the call context of the enqueued call generated by the oracle matches our request.
-        // We are issuing a regular call which is not delegate, static, or deployment. We also constrain
-        // the msg_sender in the nested call to be equal to our address, and the execution context address
-        // for the nested call to be equal to the address we actually called.
-        assert(item.public_inputs.call_context.is_delegate_call == false);
+        // We must constrain the msg_sender in the nested call to be equal to our address
+        assert(item.public_inputs.call_context.is_delegate_call == is_delegate_call);
         assert(item.public_inputs.call_context.is_static_call == is_static_call);
         assert(item.public_inputs.call_context.is_contract_deployment == false);
-        assert(
-            item.public_inputs.call_context.msg_sender.eq(self.inputs.call_context.storage_contract_address)
-        );
-        assert(item.public_inputs.call_context.storage_contract_address.eq(contract_address));
+
+        if (is_delegate_call) {
```

Comment:
We are checking these in the kernel circuit. Not sure if we should check them here. I guess it's good to early out while simulating the calls. But it should not happen if the PXE behaves. And it's extra constrains and duplicated code that we need to maintain. So maybe we can remove the checks for call context here.